src/gluonts/mx/block/scaler.py

```diff
@@ -295,4 +295,5 @@ def compute_scale(
         Tensor
             shape (N, C), identically equal to 1.
         """
-        return F.ones_like(data).mean(axis=self.axis)
+        _ones = F.ones_like(data) * F.ones_like(observed_indicator)
+        return _ones.mean(axis=self.axis)
```

Comment:
All scalers use `data` anyway (see `hybrid_forward`) so maybe this could just do

```python
return F.ones_like(observed_indicator).mean(axis=self.axis)
```

Reply:
Makes sense. Changed.